### PR TITLE
refactor: Simplify `BorderLayerRenderer` on Skia

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): part of https://github.com/unoplatform/uno/issues/6457

## PR Type

What kind of change does this PR introduce?

- Refactoring (no functional changes, no api changes)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by MartinZikmund** [comment]
@jeromelaban @Youssef1313 to make it realistic to review, I will break down the border layer renderer refactor into per-platform PRs, this one specifically for Skia only. Some additional improvements will follow after all platforms are aligned later

**Review by Youssef1313 — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarViewBaseItemChrome.IBorderInfoProvider.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.UI.Xaml.Media;
+using Uno.UI.Xaml.Controls;
+
+namespace Microsoft.UI.Xaml.Controls;
+
+partial class CalendarViewBaseItem : IBorderInfoProvider
+{
+	Brush IBorderInfoProvider.Background
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarViewBaseItemChrome.IBorderInfoProvider.cs:13]: This code appears to be duplicated in `UpdateChrome`. Consider letting `UpdateChrome` do `((IBorderInfoProvider)this).Background` (or extract this code to a method that both will call)

File: src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarViewBaseItemChrome.IBorderInfoProvider.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	{
+		get
+		{
+			var borderBrush = GetItemBorderBrush(forFocus: false);
+			if (m_isToday && m_isSelected && GetItemInnerBorderBrush() is { } selectedBrush)
+			{
+				// We don't support inner border yet, so even if not optimal we just use it as border.
+				borderBrush = selectedBrush;
+			}
+			return borderBrush;
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarViewBaseItemChrome.IBorderInfoProvider.cs:52]: Same here

File: src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.cs [hunk 3]
```diff
@@ -1,84 +1,104 @@
...
 
-		// The ellipse can only grow up to twice the available length.
-		// This is further limited by the ratio between the corner-radius
-		// of that corner and the adjacent corner on the same line.
-		var hRatio = hComplement == 0 ? 1 : (cr0 / (cr0 + hComplement));
-		var vRatio = vComplement == 0 ? 1 : (cr0 / (cr0 + vComplement));
+	/// <summary>
+	/// Updates the border.
+	/// </summary>
+	internal void Update()
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.cs:42]: Let's make sure to validate screenshot changes as well.

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.cs:36]: Done and ok!

**Pull request merged**